File: Objc/Cardholder/TPDCardholderFormExample/TPDCardholderFormExample/ViewController.m
//
//  ViewController.m
//  TPDCardholderFormExample
//
//  Created by Kevin Kao on 2025/11/5.
//

#import "ViewController.h"
#import <TPDirect/TPDirect.h>

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIView *cardholderFormView;
@property (strong, nonatomic) TPDCardholderForm   *tpdCardholderForm;
@property (weak, nonatomic) IBOutlet UIButton *payButton;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.tpdCardholderForm = [TPDCardholderForm setupWithContainer:self.cardholderFormView];
    
    [self.tpdCardholderForm setErrorColor:[self colorWithRGB:@"FF5858" withAlpha:1]];
    
    [self.tpdCardholderForm onFormUpdated:^(TPDStatus * _Nonnull status) {
            [self.payButton setEnabled:[status isCanGetCardholderPrime]];
    }];
}

- (IBAction)doneAction:(id)sender {
    
    [[[self.tpdCardholderForm onSuccessCallback:^(NSString * _Nullable prime, NSInteger status, NSString * _Nonnull message) {
                NSLog(@"prime: %@, status: %d, message: %@", prime, status, message);
        }] onFailureCallback:^(NSInteger status, NSString * _Nonnull message) {
            NSLog(@"status: %d, message: %@", status, message);
        }] getCardholderPrime];
}

- (UIColor *)colorWithRGB:(NSString *)rgbString withAlpha:(CGFloat)alpha {
    
    NSScanner *scanner = [NSScanner scannerWithString:[rgbString lowercaseString]];
    unsigned int baseColor;
    [scanner scanHexInt:&baseColor];
    
    CGFloat red   = ((baseColor & 0xFF0000) >> 16) / 255.0f;
    CGFloat green = ((baseColor & 0x00FF00) >> 8)  / 255.0f;
    CGFloat blue  =  (baseColor & 0x0000FF) / 255.0f;
    
    return [UIColor colorWithRed:red green:green blue:blue alpha:alpha];
    
}


@end

File: Swift/Cardholder/TPDCardholderFormExample/TPDCardholderFormExample/ViewController.swift
//
//  ViewController.swift
//  TPDCardholderFormExample
//
//  Created by Kevin Kao on 2025/11/5.
//

import UIKit
import TPDirect

class ViewController: UIViewController {

    @IBOutlet weak var cardholderFormView: UIView!
    @IBOutlet weak var payButton: UIButton!
    
    var tpdCardholderForm : TPDCardholderForm!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        self.tpdCardholderForm = TPDCardholderForm.setup(withContainer: self.cardholderFormView)
        
        self.tpdCardholderForm.setErrorColor(colorWithRGB(rgbString: "D62D20", alpha: 1.0))
        
        self.tpdCardholderForm.setShowEmailField(false);
        self.tpdCardholderForm.setShowPhoneNumberField(false)
        
        self.tpdCardholderForm.onFormUpdated { status in
            weak var weakSelf = self
    
            weakSelf?.payButton.isEnabled = status.isCanGetCardholderPrime()
            weakSelf?.payButton.alpha     = (status.isCanGetCardholderPrime()) ? 1.0 : 0.25
        }
    }

    
    @IBAction func doneAction(_ sender: Any) {
        self.tpdCardholderForm.onSuccessCallback { prime, status, message in
            print("prime: \(prime), status: \(status), message: \(message)")
        }.onFailureCallback { status, message in
            print("status: \(status), message: \(message)")
        }.getCardholderPrime();
    }
    
    func colorWithRGB(rgbString:String!, alpha:CGFloat!) -> UIColor! {
        
        let scanner = Scanner.init(string: rgbString.lowercased())
        var baseColor:UInt64 = UInt64()
        scanner.scanHexInt64(&baseColor)
        
        let red     = ((CGFloat)((baseColor & 0xFF0000) >> 16)) / 255.0
        let green   = ((CGFloat)((baseColor & 0xFF00) >> 8)) / 255.0
        let blue    = ((CGFloat)(baseColor & 0xFF)) / 255.0
        
        return UIColor.init(red: red, green: green, blue: blue, alpha: alpha)
    }
}

